pkg/agent/controller/egress/egress_controller.go

```diff
@@ -466,6 +551,29 @@ func (c *EgressController) syncEgress(egressName string) error {
 		return err
 	}
 
+	if egress.Spec.ExternalIPPool != "" && c.cluster.LocalNodeJoined() {
+		if c.cluster.ShouldSelectEgress(egress) {
+			if !c.localIPDetector.IsLocalIP(egress.Spec.EgressIP) {
+				a := NewNodeEgressIPAssigner(c.cluster.NodeConf)
```

Comment:
It's strange that the assigner is initialized once everytime when it's called while it's a struct. Cann't it just be a function if there's no cache need to keep?

Reply:
done